modules/scheduler/scheduler/src/main/java/org/motechproject/scheduler/builder/SchedulableJobBuilder.java

```diff
@@ -101,7 +128,21 @@ private static SchedulableJob buildRunOnceSchedulableJob() {
     }
 
     private static DateTime getEndDate(Trigger trigger) {
-        return new DateTime(trigger.getEndTime());
+        return trigger.getEndTime() == null ? null : new DateTime(trigger.getEndTime());
+    }
+
+    private static Time getTime(String cron) {
+        String[] parts = cron.split(" ");
+        return new Time(Integer.valueOf(parts[2]), Integer.valueOf(parts[1]));
+    }
+
+    private static List<DayOfWeek> getDays(String cron) {
+        String[] days = cron.split(" ")[5].split(",");
+        List<DayOfWeek> daysList = new ArrayList<>();
+        for (String day : days) {
+            daysList.add(DayOfWeek.getDayOfWeek(Integer.valueOf(day)));
+        }
+        return daysList;
```

Comment:
For the above methods, can we use some simple library/cron util to do the parsing for us?